flow-model-generator/modules/flow-model-generator-core/src/main/java/io/ballerina/flowmodelgenerator/core/CodeAnalyzer.java

```
@@ -322,77 +324,80 @@ public void visit(RemoteMethodCallActionNode remoteMethodCallActionNode) {
-                                moduleVariableDeclarationNode);
+                FunctionDefinitionNode funcDef = (FunctionDefinitionNode) member;
+                if (funcDef.functionName().text().equals("init")) {
+                    for (StatementNode statement : ((FunctionBodyBlockNode) funcDef.functionBody()).statements()) {
+                        if (statement.kind() != SyntaxKind.ASSIGNMENT_STATEMENT) {
+                            continue;
+                        }
+                        AssignmentStatementNode assignmentStmtNode = (AssignmentStatementNode) statement;
+                        if (!assignmentStmtNode.varRef().toSourceCode().trim()
+                                .equals(expressionNode.toSourceCode())) {
+                            continue;
+                        }
+                        ImplicitNewExpressionNode newExpressionNode = getNewExpr(assignmentStmtNode.expression());
+                        genAgentData(newExpressionNode, classSymbol);
                     }
-                    ImplicitNewExpressionNode newExpressionNode = getNewExpr(optInitializer.get());
-                    genAgentData(newExpressionNode, classSymbol.get());
-                    break;
                 }
             }
         } else {
-            startNode(NodeKind.REMOTE_ACTION_CALL, expressionNode.parent());
-            setFunctionProperties(functionName, expressionNode, remoteMethodCallActionNode, functionSymbol,
-                    classSymbol.get().getName().orElseThrow());
+            for (Symbol moduleSymbol : semanticModel.moduleSymbols()) {
+                if (moduleSymbol.kind() != SymbolKind.VARIABLE) {
+                    continue;
+                }
+                VariableSymbol variableSymbol = (VariableSymbol) moduleSymbol;
+                if (!variableSymbol.getName().orElseThrow().equals(expressionNode.toSourceCode())) {
```

Do we need to throw an exception for having `null` name?

---

This was existing code. I just moved the code. Lets keep it as it is no avoid regression issues